x/pairing/keeper/msg_server_relay_payment.go

```diff
@@ -267,11 +267,23 @@ func (k msgServer) RelayPayment(goCtx context.Context, msg *types.MsgRelayPaymen
 
 		// if this returns an error it means this is legacy consumer
 		if !legacy {
-			err = k.projectsKeeper.AddComputeUnitsToProject(ctx, clientAddr.String(), uint64(relay.Epoch), relay.CuSum)
+			err = k.projectsKeeper.ChangeComputeUnitsToProject(ctx, clientAddr.String(), uint64(relay.Epoch), relay.CuSum)
+
 			if err != nil {
 				details["error"] = err.Error()
 				return errorLogAndFormat("relay_payment_failed_project_add_cu", details, "Failed to add CU to the project")
 			}
+
+			project, _, err := k.projectsKeeper.GetProjectForDeveloper(ctx, clientAddr.String(), uint64(ctx.BlockHeight()))
+			if err != nil {
+				details["error"] = err.Error()
+				return errorLogAndFormat("relay_payment_failed_getting_project", details, "Failed to get project for developer")
+			}
+			err = k.subscriptionKeeper.ChangeComputeUnitsToSubscription(ctx, project.GetSubscription(), relay.CuSum)
```

Comment:
same here

Reply:
@Yaroms , same here regarding "Charge..." and not "Change" (instead of "Add...").

Reply:
done